From 17508b3e3a8991ecc0cf767ad2302d34141b8616 Mon Sep 17 00:00:00 2001
From: Aashvij Shenai <a-shenai@ti.com>
Date: Fri, 29 Mar 2024 15:40:53 +0530
Subject: [PATCH] am62px: Change DMA allocation

This patch removes the DMA channel allocations fo A53 and MCU cores so
that the first channels are allocated to DM R5. This will be used to
demonstrate the use of High Capacity BCDMA channels. In Am62Px, there
are 4 HC channels
---
 .../am62px/sciclient_defaultBoardcfg_rm.c     | 42 ++++---------------
 1 file changed, 9 insertions(+), 33 deletions(-)

diff --git a/source/drivers/sciclient/sciclient_default_boardcfg/am62px/sciclient_defaultBoardcfg_rm.c b/source/drivers/sciclient/sciclient_default_boardcfg/am62px/sciclient_defaultBoardcfg_rm.c
index a22e6f6b4459..05d228843932 100644
--- a/source/drivers/sciclient/sciclient_default_boardcfg/am62px/sciclient_defaultBoardcfg_rm.c
+++ b/source/drivers/sciclient/sciclient_default_boardcfg/am62px/sciclient_defaultBoardcfg_rm.c
@@ -96,7 +96,7 @@ __attribute__(( aligned(128), section(".boardcfg_data") )) =
                 .magic = TISCI_BOARDCFG_RM_RESASG_MAGIC_NUM,
                 .size = (uint16_t) sizeof(struct tisci_boardcfg_rm_resasg),
             },
-            .resasg_entries_size = 123 * sizeof(struct tisci_boardcfg_rm_resasg_entry),
+            .resasg_entries_size = 119 * sizeof(struct tisci_boardcfg_rm_resasg_entry),
         },
     },
     .resasg_entries = {
@@ -154,34 +154,22 @@ __attribute__(( aligned(128), section(".boardcfg_data") )) =
             .start_resource = 0,
             .host_id = TISCI_HOST_ID_ALL,
         },
-        {
-            .num_resource = 18,
-            .type = TISCI_RESASG_UTYPE (TISCI_DEV_DMASS0_BCDMA_0, TISCI_RESASG_SUBTYPE_BCDMA_RING_BLOCK_COPY_CHAN),
-            .start_resource = 0,
-            .host_id = TISCI_HOST_ID_A53_2,
-        },
         {
             .num_resource = 6,
             .type = TISCI_RESASG_UTYPE (TISCI_DEV_DMASS0_BCDMA_0, TISCI_RESASG_SUBTYPE_BCDMA_RING_BLOCK_COPY_CHAN),
-            .start_resource = 18,
+            .start_resource = 0,
             .host_id = TISCI_HOST_ID_WKUP_0_R5_0,
         },
         {
             .num_resource = 6,
             .type = TISCI_RESASG_UTYPE (TISCI_DEV_DMASS0_BCDMA_0, TISCI_RESASG_SUBTYPE_BCDMA_RING_BLOCK_COPY_CHAN),
-            .start_resource = 18,
+            .start_resource = 0,
             .host_id = TISCI_HOST_ID_WKUP_0_R5_1,
         },
         {
-            .num_resource = 2,
-            .type = TISCI_RESASG_UTYPE (TISCI_DEV_DMASS0_BCDMA_0, TISCI_RESASG_SUBTYPE_BCDMA_RING_BLOCK_COPY_CHAN),
-            .start_resource = 24,
-            .host_id = TISCI_HOST_ID_MCU_0_R5_0,
-        },
-        {
-            .num_resource = 6,
+            .num_resource = 26,
             .type = TISCI_RESASG_UTYPE (TISCI_DEV_DMASS0_BCDMA_0, TISCI_RESASG_SUBTYPE_BCDMA_RING_BLOCK_COPY_CHAN),
-            .start_resource = 26,
+            .start_resource = 6,
             .host_id = TISCI_HOST_ID_ALL,
         },
         {
@@ -238,34 +226,22 @@ __attribute__(( aligned(128), section(".boardcfg_data") )) =
             .start_resource = 52,
             .host_id = TISCI_HOST_ID_ALL,
         },
-        {
-            .num_resource = 18,
-            .type = TISCI_RESASG_UTYPE (TISCI_DEV_DMASS0_BCDMA_0, TISCI_RESASG_SUBTYPE_BCDMA_BLOCK_COPY_CHAN),
-            .start_resource = 0,
-            .host_id = TISCI_HOST_ID_A53_2,
-        },
         {
             .num_resource = 6,
             .type = TISCI_RESASG_UTYPE (TISCI_DEV_DMASS0_BCDMA_0, TISCI_RESASG_SUBTYPE_BCDMA_BLOCK_COPY_CHAN),
-            .start_resource = 18,
+            .start_resource = 0,
             .host_id = TISCI_HOST_ID_WKUP_0_R5_0,
         },
         {
             .num_resource = 6,
             .type = TISCI_RESASG_UTYPE (TISCI_DEV_DMASS0_BCDMA_0, TISCI_RESASG_SUBTYPE_BCDMA_BLOCK_COPY_CHAN),
-            .start_resource = 18,
+            .start_resource = 0,
             .host_id = TISCI_HOST_ID_WKUP_0_R5_1,
         },
         {
-            .num_resource = 2,
-            .type = TISCI_RESASG_UTYPE (TISCI_DEV_DMASS0_BCDMA_0, TISCI_RESASG_SUBTYPE_BCDMA_BLOCK_COPY_CHAN),
-            .start_resource = 24,
-            .host_id = TISCI_HOST_ID_MCU_0_R5_0,
-        },
-        {
-            .num_resource = 6,
+            .num_resource = 26,
             .type = TISCI_RESASG_UTYPE (TISCI_DEV_DMASS0_BCDMA_0, TISCI_RESASG_SUBTYPE_BCDMA_BLOCK_COPY_CHAN),
-            .start_resource = 26,
+            .start_resource = 6,
             .host_id = TISCI_HOST_ID_ALL,
         },
         {
-- 
2.34.1

